### ISSUE TITLE
Call NameReference Appearance?

### ISSUE THREAD

**Issue by mdoering** [CatalogueOfLife/coldp#41]
Appearance of a name within a treatment/article is a term already been used by http://www.nomencurator.org/
which seems much like we use NameReference now

**Comment by deepreef** [comment]
[Here](https://watermark.silverchair.com/j.1095-8312.2001.tb01348.x.pdf?token=AQECAHi208BE49Ooan9kkhW_Ercy7Dm3ZL_9Cf3qfKAc485ysgAAAuYwggLiBgkqhkiG9w0BBwagggLTMIICzwIBADCCAsgGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMpA0l450vHNdZWCgmAgEQgIICmSIfrlY5V3MMueGbIjC80OxywdVoBmNyh88QrW1JuJJ8dgSKtmHqGsREJQte-WhZbVm051Z1gl218kCJQZ3rE0PGbV2eNz5toBE4Gea1vFdmDuCC_QKTXUGPQhREgjz-fxLtgUC9QIBn9b83XmEVQ2L1GdmYRrETXkcpHdWSurF-qVWHcUjlb3RO-spxJwHCqB0OyxXs_1KZrdFHai2jWvWAW1IHFp3_wqK8uLJ939XH0o41ZWIJnxJSzCqidKmyG6aAHcQGQPQCFQdpT8pAUV_nXIkKARHuevObIRqew-9GbtpMGVNUhyajkdxdlgLu6nhHCbP9UktgzR8WBzeKTMb2e4ukUZ3Vjpzl8zyf5K5hcQ-VbD7trU8PbbolZ8_USVaH7mLwKwsskcCiOG-aAfv3kn5Ouy4LPkHB0IVZ9gx5UfK0o8TIjh41UKk991d08SwU8X9iDNqvZBhr33Aw2WYgEkmgqM5PUTDMz_3o9tja9_TCVIP6A5rRNYu9NULxhoC_4O9XrEvcKeHloC9Sih72Sz2PHi4fru8waNItNMniyKfkAF_lfbNqZ28l0i_4r8eQxHPStx-ZfcAqt6-iMY2_PXRri2rfoDyhzZEZa5TM-2nRQap-5Ad9t-2BlZ5U31HK76G5Kp8Edk-eXOfigsEGlMOfVcyCrhimu-9q4J9v85_uJYBFeezBEqSUgvFc7ZT0XhAnFyKks1qvkEr0KlZ4LTYCGPd8PDR3fvpH7cA3g09I765FqfmvM8Afs0Lrzh1oEGjVu7YX4QE0023EOzZReQnJ2Vkc1r49a3huOeGuXabj7kG04q98tqMCgK7BxZYNm9gEbgxuSGGUDYEJd72GUEnuXGakuWQb7_wxTrNylq5Pv9a7wKJ2) is a published version.  Not sure this link will work because the hyperlink requires a huge token that may expire. Can also search for "Nomencurator" on Google Scholar.

**Comment by deepreef** [comment]
Here is the PDF:
[j.1095-8312.2001.tb01348.x.pdf](https://github.com/CatalogueOfLife/coldp/files/5517708/j.1095-8312.2001.tb01348.x.pdf)

**Comment by mdoering** [comment]
@dhobern @gdower what do you think, should we rename NameReference to NameAppearance or even just Appearance or rather keep it "relational style"?

**Comment by deepreef** [comment]
I would only recommend adopting Ytow's term (either "NameAppearance" or just "Appearance") if you match is definition of the instance.  If there is a 1:1 relationship between a TaxonName instance and a Reference instance, then you should call it a TaxonomicNameUsage.  If a single Reference instance might have many instances of NameReference (e.g., if the text-string name appears five times on page 12 of the Reference, three times on page 13 of the reference and once on page 32 of the Reference, and you therefore have a total of nine NameReference instances for the same Reference), then you should adopt (some variant of) Ytow's term.  If you have some other basis for establishing instances of NameReference that is neither 1:1, nor complete for all text-string appearances of the name within the Reference, then you might want to call it something else.

The definition of the instance should dictate what term you use.

**Comment by mdoering** [comment]
This is still open. I originally thought the name _NameReference_ is more familiar to database savy people. But I also encountered the first problems when people used NameReference exactly as that, the many to many table bridging names and references, thereby bypassing the core Name.referenceID field.

I now feel we are in better shape to remove this entity alltogether or call it `Appearance` so people are required to read the documentation on what it exactly means. @dhobern any strong feelings?

**Comment by deepreef** [comment]
Just to make sure there is no ambiguity here, a TaxonomicNameUsage (TNU) instance represents the intersection of a name-as-object (i.e., Protonym) and a Reference.  If the Reference is Smith 1950, and the name-as-object is Aus bus L., then there is one TNU representing Aus bus L. sec. Smith 1950.  However, when we read the treatment of Aus bus L. within Smith 1950, there may be 15 different times in the text where some variant of "Aus bus L." appears as literal text.  Maybe 3 of them are "Aus bus L.", 2 of them are "Aus bus", and 10 of them are "A. bus". Four of them may appear on page 16, 6 of them may appear on page 17, and 5 of them may appear on page 18.  Thus, for the single TNU of Aus bus L. sec Smith 1950, there are 15 Appearance instances, scattered among three different pages, and represented by three different text strings (or two different text strings if you ignore the "L." part of "Aus bus L." vs. "Aus bus").

I'm not quite sure if a coldp instance of *NameReference* represents a TNU, or an Appearance, or something in-between.  If it's something in-between (i.e., there are more than 1 instances associated with the treatment of Aus bus L. sec. Smith 1950, but less than 15), then I would still go with "Appearance" and assume the same granularity as defined by Ytow, but not capturing all 15 instances individually.

I hope that makes sense!

**Comment by mdoering** [comment]
A ColDP TNU would be a `NameUsage` or `Taxon` record with the former being a convenient 1:1 combination of a `Name` and a `Taxon` record. Taxon/NameUsage carries the TNU reference via `accordingToID`

[NameReference](https://github.com/CatalogueOfLife/coldp/blob/master/README.md#namereference) here really is an Appearance sensu Ytow. The description currently says:

> References are usually classic bibliographic citations on the article level. In many cases it is desirable to point and link to specific pages, e.g. where exactly a name has first been published.
> 
> In order to avoid highly redundant references ColDP allows to share NameReference records that are microcitations to a specific page where a name was mentioned that belongs to a specific Reference record. They are appearances of a name in a publication.
> 
> NameReference should not be used to designate the original publication of the name which should rather be done using Name.referenceID. Using NameReference is optional and supplements Name.referenceID.

**Comment by deepreef** [comment]
Ok, thanks... but I guess I'm a little unclear on:

>NameReference records that are microcitations to a specific page where a name was mentioned

This makes it sound like there would be only one NameReference instance per page; when in fact there may be many Appearances for the same name on a single page.  There's no standard way to provide an "address" to a position on a page that is more granular than the page as a whole, so I don't know if there are 3 Appearances on a single page whether there would be as many as three NameReference instances for that name on that single page, or if there would be only one NameReference instance for that name/page, with some property like "numberOfAppearances=3"; or if there would be only one NameReference instance with no quantity property, with an implication that "this name appears *at least once* on this page within this reference".

It probably doesn't matter, because any one of these three scenarios would probably still be appropriate to use the term "Appearance".  But I just wanted to make sure I understood the cardinality between NameReference sensu coldb and Appearance sensu Ytow.

Sorry for the digression!

**Comment by mdoering** [comment]
It is a practical thing that allows taxonomists to share exact links to points of interest in a publication, together with a remark whats so special about it. Could be an image, the protologue or any other interesting bit thats worth providing direct links to. 

Maybe we should highlight the linking aspect in the name of that entity. Without a link or at least a remark why its special its pretty useless. In that light it is not exactly like Ytow, the name might not always appear in that linked spot.

I was reading @dhobern mails when he originally proposed this as `PageReference`. Maybe that was a better name after all. 


> Redefine PageReference as follows:
> - nameID (mandatory) - Pointer to the name used on the page in the reference. Refers to an existing Name.ID within this data package.
> - referenceID (mandatory) - Pointer to the reference that includes this page. Refers to an existing Reference.ID within this data package.
> - page - The exact page the microcitation is pointing to within the reference.
> - link - A URL link to the exact page of the reference. If only a link for the entire reference is available this should only be included in the main reference, not here again.
> - remarks - Opportunity to indicate the context, e.g. "Plate VIII: Genitalia drawings" or "Original description of Abies", etc.
> 
> In other words, the PageReference becomes a set of annotations on names rather than something that needs to be referenced from inside Name, Taxon, etc. It allows complete bibliographies to be exposed for any species, and these can then be aggregated at the Taxon level via synonymy. It avoids the need for us to do have separate versions that use taxonID in place of nameID, etc.
>

**Comment by deepreef** [comment]
Yes, `PageReference` might be better in this context, but that limits you to the granularity of a page.  Many modern (electronic) publications don't even make pages any more (one page for the whole article), and I'm sure there are many cases where more than one point of interest for a given name on a given page.  But I guess if it's really just an index of of which pages contain interesting information about which names, then `PageReference` (plus some characterization of what makes it interesting -- e.g.., photo, key, type fixation, treatment start, synonymy, etc.) might make more sense.

**Comment by mdoering** [comment]
We have decided to remove NameReference from the first version of ColDP and stage it for the next version. It is not supported by the CLB importer yet, the name is not in agreement yet and it confused data publishers for COL who used the table wrongly. 

<img width="265" alt="Screenshot 2021-04-19 at 17 47 11" src="https://user-images.githubusercontent.com/327505/115265222-50743b80-a137-11eb-8374-d1d1f4e4b33d.png">
